lib/overcommit/hook_context/pre_push.rb

```diff
@@ -18,9 +18,21 @@ def pushed_refs
     end
 
     def modified_files
-      @modified_files ||= Overcommit::GitRepo.modified_files(
-        refs: "#{pushed_refs[0].remote_sha1}..#{pushed_refs[0].local_sha1}"
-      )
+      @modified_files ||= Overcommit::GitRepo.modified_files(refs: refs)
+    end
+
+    def modified_lines_in_file(file)
+      @modified_lines ||= {}
+      unless @modified_lines[file]
+        @modified_lines[file] =
+          Overcommit::GitRepo.extract_modified_lines(file, refs: refs)
+      end
+
+      @modified_lines[file]
```

Comment:
You can shorten this method implementation to:

```ruby
@modified_lines ||= {}
@modified_lines[file] ||= Overcommit::GitRepo.extract_modified_lines(...)
```

Reply:
Awesome.